src/sysinfo/nvapi_adapter.cpp

```diff
@@ -211,27 +228,39 @@ namespace dxvk {
         if (m_vkProperties.deviceID >= 0x2600)
             return NV_GPU_ARCHITECTURE_AD100;
 
+        // See https://gitlab.freedesktop.org/mesa/mesa/-/blob/main/src/nouveau/vulkan/nvk_physical_device.c
+        // for NVK properties and features
+
         // KHR_fragment_shading_rate's
         // primitiveFragmentShadingRateWithMultipleViewports is supported on
         // Ampere and newer
+        // TODO: We haven't found a way yet to identify Ampere on NVK
         if (IsVkDeviceExtensionSupported(VK_KHR_FRAGMENT_SHADING_RATE_EXTENSION_NAME)
             && m_vkFragmentShadingRateProperties.primitiveFragmentShadingRateWithMultipleViewports)
             return NV_GPU_ARCHITECTURE_GA100;
 
-        // Variable rate shading is supported on Turing and newer
-        if (IsVkDeviceExtensionSupported(VK_NV_SHADING_RATE_IMAGE_EXTENSION_NAME))
+        // Mesh shader is supported on Turing and newer
+        // This needs https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/27196 on NVK
+        if (IsVkDeviceExtensionSupported(VK_EXT_MESH_SHADER_EXTENSION_NAME))
```

Comment:
May be `VK_KHR_fragment_shader_barycentric` is better since that is already implemented in NVK (https://gitlab.freedesktop.org/mesa/mesa/-/blob/main/src/nouveau/vulkan/nvk_physical_device.c?ref_type=heads#L101)? Alternatively `VK_KHR_draw_indirect_count`?

Reply:
`VK_KHR_fragment_shader_barycentric`, yes, we could use that as well. `VK_KHR_draw_indirect_count`, not really unless we guard this behind `HasNvkDriver` because [NV proprietary driver seems to support this on older architectures as well](https://vulkan.gpuinfo.org/listdevicescoverage.php?extension=VK_KHR_draw_indirect_count&platform=all).

Reply:
I went for `VK_KHR_fragment_shader_barycentric`.